crates/aptos/src/node/mod.rs

```diff
@@ -231,6 +235,59 @@ impl ValidatorNetworkAddressesArgs {
     }
 }
 
+#[derive(Parser)]
+pub struct GetPoolAddress {
+    #[clap(long)]
+    pub(crate) owner_address: AccountAddress,
+    #[clap(long)]
+    pub(crate) operator_address: Option<AccountAddress>,
+    #[clap(long)]
+    pub(crate) employee_account_index: Option<u64>,
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+}
+
+#[async_trait]
+impl CliCommand<AccountAddress> for GetPoolAddress {
+    fn command_name(&self) -> &'static str {
+        "GetPoolAddress"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<AccountAddress> {
+        let owner_address = self.owner_address;
+
+        let client = self.rest_options.client(&self.profile_options.profile)?;
+        let stake_pool_exists = client
+            .get_account_resource_bcs::<StakePool>(owner_address, "0x1::stake::StakePool")
+            .await
+            .is_ok();
+        let stake_pool_address = if stake_pool_exists {
+            owner_address
```

Comment:
This doesn't really tell you if the address is a delegated pool or the original, but I guess that works.